samples/solidity/contracts/ERC721WithData.sol

```diff
@@ -76,6 +105,23 @@ contract ERC721WithData is Context, Ownable, ERC721, IERC721WithData {
     }
 
     function _baseURI() internal view virtual override returns (string memory) {
-        return 'firefly://token/';
+        bytes memory tempURITest = bytes(_baseTokenURI);
+        if (tempURITest.length == 0) {
```

Comment:
I don't know Solidity well enough, but is this really the only way to check if a string is set/unset?